### PR TITLE
[conductor] Fix branch version to use framework branch

### LINKED ISSUE #101773: [conductor] Release candidate branch version is using version instead of branch
See https://github.com/flutter/flutter/pull/101771

### PR DESCRIPTION
**Pull request by CaseyHillers**
Fixes https://github.com/flutter/flutter/issues/101773

### CODE REVIEW

File: dev/conductor/core/lib/src/next.dart [hunk 1]
```diff
@@ -194,10 +194,10 @@ class NextContext extends Context {
           previousCheckoutLocation: state.framework.checkoutPath,
         );
         stdio.printStatus('Writing candidate branch...');
-        bool needsCommit = await framework.updateCandidateBranchVersion(state.releaseVersion);
+        bool needsCommit = await framework.updateCandidateBranchVersion(state.framework.candidateBranch);
```
Comment (CaseyHillers) [inline dev/conductor/core/lib/src/next.dart:197]: I'm not really sure how to test this. Any feedback would be helpful, but i'm not sure why the version was coming out as the branch in the test.

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:197]: looking now....

Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart:197]: Thanks! I did `expect(stdio.stdout, contains('release-candidate-branch.version containing $candidateBranch'));` to follow the format of the existing tests so if we refactor it's a bit easier.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**